k8s/agent-scaler/agent-scaler.sh

```diff
@@ -6,23 +6,35 @@ check_delay=${CHECK_DELAY:-10}
 # Kubernetes 1.21 supports jobs TTL controller, which cleans up jobs automatically
 # see https://kubernetes.io/docs/concepts/workloads/controllers/ttlafterfinished/
 use_k8s_ttl_controller=${USE_K8S_TTL_CONTROLLER:-false}
+restart_agents_on_boot=${RESTART_AGENTS_ON_BOOT:-false}
 
 namespace=$(cat /var/run/secrets/kubernetes.io/serviceaccount/namespace)
 
+if [ -z "${AGENT_TAG}" ]; then
+    # if AGENT_TAG is not set through --build-arg,
+    # use the default agent image and tag
+    export AGENT_TAG="foundationdb/joshua-agent:latest"
+fi
+
+if [ $restart_agents_on_boot == true ]; then
+    # mark existing jobs to exit after the current test completes
+    kubectl -n ${namespace} label pods -l app=joshua-agent last_test=true --overwrite=true
+fi
+
 # run forever
 while true; do
 
     if [ $use_k8s_ttl_controller == false ] ; then
       # cleanup finished jobs (status 1/1)
-      for job in $(kubectl get jobs -n "${namespace}" | grep -E -e 'joshua-agent-[0-9-]*\s*1/1' | awk '{print $1}'); do
+      for job in $(kubectl get jobs -n "${namespace}" --no-headers | grep -E -e 'joshua-agent-[0-9-]*\s*1/1' | awk '{print $1}'); do
```

Comment:
Doing a proper label selection and using jsonpath output might be the better solution to using grep + awk, e.g. https://kubernetes.io/docs/reference/kubectl/jsonpath.

Was there any issue with the headers? The grep command should filter them out anyway?